.github/workflows/build-release-doc-pages.yml

```diff
@@ -0,0 +1,29 @@
+name: Generate github pages
+
+on:
+  push: 
+    branches: [ feature/documentation-pages, main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: feature/documentation-pages
+
+      - name: Setup Maven
+        uses: stCarolas/setup-maven@v4.1
+
+      - name: Generate Maven site
```

Comment:
No need it. Just using official `setup java`
https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven